client/keys.go

```diff
@@ -445,7 +445,15 @@ func (hw *httpWatcher) Next(ctx context.Context) (*Response, error) {
 // provided endpoint's path to the root of the keys API
 // (typically "/v2/keys").
 func v2KeysURL(ep url.URL, prefix, key string) *url.URL {
-	ep.Path = path.Join(ep.Path, prefix, key)
+	// It concatenates all parts together manually. It cannot use
```

Comment:
We .... We cannot use... it does not preserve the trailing slash. We call CanonicalURL to further cleanup the path.